extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/RestDataProcessor.java

```diff
@@ -57,4 +63,36 @@ JsonbSerializerBuildItem registerJsonbSerializers() {
     RuntimeInitializedClassBuildItem el() {
         return new RuntimeInitializedClassBuildItem(EL.class.getCanonicalName());
     }
+
+    private void verifyHalResources(IndexView index, List<RestDataResourceBuildItem> resourceBuildItems) {
+        if (!hasJsonProcessor() && hasHalResource(index, resourceBuildItems)) {
+            logger.warn("Cannot generate HAL endpoints without a serializer. "
+                    + "Make sure you have either Jsonb or Jackson RESTEasy extension enabled.");
+            throw new RestDataBuildException("Cannot generate HAL endpoints without a serializer");
```

Comment:
Either you log a warning or you throw an exception. In this case, I would just throw an exception with the proper message. 

Also I don't think you need a specific exception for this. An `IllegalStateException` would be perfectly adequate IMHO.